apps/melding-form/src/middleware.test.ts

```diff
@@ -0,0 +1,49 @@
+import { NextRequest, NextResponse } from 'next/server'
+
+import { middleware } from './middleware'
+
+vi.mock('next/server', async () => {
+  const actual = await vi.importActual<typeof import('next/server')>('next/server')
+  return {
+    ...actual,
+    NextResponse: {
+      redirect: vi.fn((url: URL) => ({ type: 'redirect', url: url.toString() })),
+      next: vi.fn(() => ({ type: 'next' })),
+    },
+  }
+})
+
+const createMockRequest = (cookies: Record<string, string>, url = 'http://localhost/test'): NextRequest =>
+  ({
+    cookies: {
+      get: (key: string) => (cookies[key] ? { value: cookies[key] } : undefined),
+    },
+    url,
+  }) as NextRequest
+
+describe('middleware', () => {
+  afterEach(() => {
+    vi.clearAllMocks()
+  })
+
+  it('redirects to / if token is missing', () => {
+    const request = createMockRequest({ id: '123' })
+    const result = middleware(request)
+    expect(NextResponse.redirect).toHaveBeenCalledWith(new URL('/', request.url))
```

Comment:
We generally add a newline between variable declarations and `expect` statements in tests. I think that makes it a bit more readable.